raft/src/main/java/org/apache/kafka/raft/internals/KafkaRaftMetrics.java

```diff
@@ -87,17 +89,34 @@ public KafkaRaftMetrics(Metrics metrics, String metricGrpPrefix, QuorumState sta
         this.currentLeaderIdMetricName = metrics.metricName("current-leader", metricGroupName, "The current quorum leader's id; -1 indicates unknown");
         metrics.addMetric(this.currentLeaderIdMetricName, (mConfig, currentTimeMs) -> state.leaderId().orElse(-1));
 
-        this.currentVotedIdMetricName = metrics.metricName("current-vote", metricGroupName, "The current voted leader's id; -1 indicates not voted for anyone");
+        this.currentVotedIdMetricName = metrics.metricName("current-vote", metricGroupName, "The current voted id; -1 indicates not voted for anyone");
         metrics.addMetric(this.currentVotedIdMetricName, (mConfig, currentTimeMs) -> {
             if (state.isLeader() || state.isCandidate()) {
                 return state.localIdOrThrow();
-            } else if (state.isVoted()) {
-                return state.votedStateOrThrow().votedId();
             } else {
-                return -1;
+                return (double) state.maybeVotedState()
```

Comment:
What don't we need the casting before?

Reply:
I am not 100% of the issue. May be Java 8 don't do implicit casting if the value is not a number literal.